### ISSUE TITLE
Deprecated View system

### ISSUE THREAD

**Issue by ZsoltBertalan** [viafoura/SDK-android#5]
This SDK uses the View system, that will be essentially deprecated soon due to Jetpack Compose. I have a few questions related to this:
* Do you plan to support Compose?
* Is it possible to use this SDK with Compose? A quick look suggests it's not, at least not without the View wrappers.
* If not, will the direct API access still supported, so we can use Compose?

We want to get rid of the Viafoura WebViews, but also want to switch to Compose this year, so this SDK is not an option for us, it seems.

**Comment by mdesimone-viafoura** [comment]
Hey @ZsoltBertalan!

_- Is it possible to use this SDK with Compose? A quick look suggests it's not, at least not without the View wrappers._

It's possible to use this SDK on a compose project. We don't support Compose directly yet, but you can embed the fragment in compose [(interop documentation)](https://developer.android.com/jetpack/compose/interop/interop-apis) or wrap it on any other wrapper component (activity/fragment). We don't have native Compose elements as of today.

- Do you plan to support Compose?

It's something that it's on our roadmap to do at some point, but it has not been prioritized as of yet. If we see there is some value on adding native compose elements or that the wrapper approach is not enough, we could bump up the priority on this. I will take this feedback with the team.

- If not, will the direct API access still supported, so we can use Compose?

Direct API access will still be supported, although we strongly encourage mobile integrations to move over to SDK.

**Comment by ZsoltBertalan** [comment]
Fair enough, thanks. Based on our experience with the WebViews we don't want to go down the wrapper route because of the overhead and the loss of control, but it's good to know that we have the API as a backup.
I think if you find value in a View based SDK, you should find value in a Compose based one too. It's not official, but it's my strong opinion, that the View system will be deprecated. Unofficially much sooner than officially.
